pkg/server/server.go

```diff
@@ -795,6 +797,15 @@ func (s *Server) ListObjects(ctx context.Context, req *openfgav1.ListObjectsRequ
 	}, nil
 }
 
+func (s *Server) ListObjects(ctx context.Context, req *openfgav1.ListObjectsRequest) (*openfgav1.ListObjectsResponse, error) {
+	err := s.CheckAuthz(ctx, req.GetStoreId(), authz.ListObjects)
+	if err != nil {
+		return nil, err
+	}
+
+	return s.ListObjectsWithoutAuthz(ctx, req)
+}
```

Comment:
Hmmm i don't know if I like how this looks architecturally speaking 🥲 Authentication is made as a middleware (https://github.com/openfga/openfga/blob/f3e9c60ee378af91c6f4e24177865961cc9b8824/internal/middleware/authn/authn.go#L12-L21) and by the time the request has made it to `server.go`, it's already been authenticated. I see no reason why authorization should be any different